src/display/api.js

```diff
@@ -1601,6 +1605,63 @@ var WorkerTransport = (function WorkerTransportClosure() {
           }, this);
       }
 
+      messageHandler.on('GetReader', function(data, sink) {
+        this._PDFNetworkStream = new PDFNetworkStream(data);
+        this._fullReader = this._PDFNetworkStream.getFullReader();
+
+        sink.onPull = () => {
+          this._fullReader.read().then(function({ value, done, }) {
+            if (done) {
+              sink.close();
+              return;
+            }
+            // Enqueue data chunk into sink, and transfer it
+            // to other side as `Transferable` object.
+            sink.enqueue(new Uint8Array(value), 1, [value]);
+          }).catch(function(reason) {
```

Comment:
No need for an intermediate function here, `}).catch(sink.error);` should work just as well.